### DISCUSSION TITLE
Improve the "access as Scene as unique Name"

### DISCUSSION THREAD

**Discussion by Proksimas** [godotengine/godot-proposals#6354] [Editor]
Hello everyone, the "access as Scene as unique Name" is a great idea and very very usefull. 
But it doesn't fix a 'user issue' that you have to go in your script, do a onready var my_variable = $"%NameOfTheUniqueNode"

Often when you have about ten nodes that you have to do this, this is annoying.
My proposal is very simple:

It is possible do automatise the onready var my_variable = $"%NameOfTheUniqueNode" directly when you check the" access as scene" and select the node which contains the script target ? Like signals ?

You just have to follow a template which split the NameNode on a user variable 
For example:
In the scene Tree:
GroupsQuests
     CurrentQuests            
     ActiveButton

1) RightClick on CurrentQuests  node and check "access as Scene as unique Name"
2) Select GroupsQuests (which have a script.gd)
3) automatically create at the beginning of the script:  onready var current_quests = $"%CurrentQuests"

What do you think ?
Cheers, 
Proksima

**Comment by dalexeev** [comment]
You can drag the node into the Script Editor. If you hold down the <kbd>Ctrl</kbd> key, the `@onready` variable will be added.

**Reply by Proksimas** [reply]
wow thanks ! 
I didn't figure that. Maybe improve in documentation that feature is possible ? (or i don't find it...) .
But it doesn't take the $"%MyNode". SO i have to add manually. 
Close perfect ;)


Thank you !

**Reply by dalexeev** [reply]
At least in 4.0 Scene Unique Names are taken into account.
Perhaps we should add a tooltip to the Scene dock so that more users know about this feature.
